src/modules/UI/Settings/reducer.js

```diff
@@ -286,9 +290,10 @@ export const settingsLegacy = (state: SettingsState = initialState, action: Acti
 
     case 'OVERWRITE_THEN_DELETE_TOKEN_SUCCESS': {
       // where oldCurrencyCode is the sender, and tokenObj.currencyCode is the receiver (new code)
-      const receiverCode = data.tokenObj.currencyCode
-      const senderCode = data.oldCurrencyCode
-      const { tokenObj } = data
+      if (!action.data) throw new Error('Invalid action')
+      const receiverCode = action.data.tokenObj.currencyCode
+      const senderCode = action.data.oldCurrencyCode
```

Comment:
Should this be checking to make sure that `tokenObj` exists?

Reply:
the point of this PR is not to fix existing bugs, but to fix bugs caused by the previous action typing PR

Reply:
the point of this PR is not to fix existing bugs, but to fix bugs caused by the previous action typing PR